src/main/java/ui/driver/DriverManager.java

```diff
@@ -0,0 +1,41 @@
+package ui.driver;
+
+import org.openqa.selenium.WebDriver;
+import ui.driverConfig.DriverProperties;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import static ui.logger.UILogger.LOGGER;
+
+public class DriverManager {
+    private final static int IMPLICIT_WAIT_TIMEOUT = 5;
+    private final static int PAGE_LOAD_TIMEOUT = 60;
+
+    private static ThreadLocal<WebDriver> ThreadLocal = new ThreadLocal<>();
+
+    private DriverManager() {
+    }
+
+    public static void setupDriver() {
+        LOGGER.info("Setting driver...");
+        WebDriver driver = DriverProperties.setUpDriverWithProperties();
```

Comment:
you call method WithProperties and then add properties for timeouts, etc

Reply:
Refactor WebDriver classes (change logic)